setup.py

```diff
@@ -570,26 +570,46 @@ def determine_gl_flags():
 
 def determine_sdl2():
     flags = {}
+    # This is redundant; the only time determine_sdl2 is called is preceded by
+    # verification that this condition is already satisfied
     if not c_options['use_sdl2']:
         return flags
 
+    # Check the environment for a Kivy SDL2 path (all platforms) or for Include
+    # and Lib (intended to catch Windows installation, see #4941)
     sdl2_path = environ.get('KIVY_SDL2_PATH', None)
 
-    if sdl2_flags and not sdl2_path and platform == 'darwin':
-        return sdl2_flags
+    # SDL2 path was not contained in the environment; attempt to infer it
+    if not sdl2_path:
+        if platform == 'darwin' and sdl2_flags:
+            return sdl2_flags
+        # no pkgconfig info, or we want to use a specific sdl2 path, so perform
+        # manual configuration
+
+        # Not on OSX, and/or no sdl2_flags defined
+        sdl2_paths = []
+        sdl2_path_candidates = []
+        # Check in the executable's include dir
+        sdl2_path_candidates.append(
+            join(dirname(sys.executable), 'include', 'SDL2'))
+        # For every path in the %INCLUDE% (Windows), add the SDL2 subdir
+        sdl2_path_candidates.extend(
+            [join(p, 'SDL2') for p in
+             environ.get('INCLUDE', '').split(os.pathsep)]
```

Comment:
I tried to speed read the issue as there was other stuff going on there. It wasn't clear to me how much the issue was the usage of build tools rather than visual studio which we don't officially support or venv. Doesn't sys.executable take care of the venv header issue? If not, we should find what the correct path is and use that.